docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,200 @@
+    tlcc mrenclave can be retrieved through ercc;
+    the validation of these parameters occurs when the registration tx is processed by ercc;
+    if necessary, a pre-validation can be performed through tlcc.
   end note
+  Enclave1 -> Enclave1: CC_Parameters <- retrieve CC_Id, CC_Version, MSPID, Channel_Id, TLCC_MREnclave
 
-  note left Enclave1
-    ECC_ID is //short// unique identifier
-    ECC_ID = Hash(ECC_S_PK)
-  end note
+  Enclave1 -> Enclave1 : generate Private_Keys <- <Enclave_SK, Enclave_DK>
+  Enclave1 -> Enclave1 : Public_Keys <- Public(Private_Keys)
 
-  Enclave1  -> ECC1 ++: binding (ECC_S_PK, ECC_E_PK)
-  ECC1 -> ECC1 : SIG_Org_1 <- create signature
-  note left
-    Org_1 takes "ownership" of this enclave
-    sign <CC_NAME, CC_VER, Org_1, ECC_S_PK, ECC_E_PK> with Org_1_S_SK
-  end note
-  return SIG_Org_1
-  |||
+  Enclave1 -> Enclave1 : Attested_Data <- <CC_Parameters, Public_Keys>
 
-  Enclave1 -> Enclave1 : generate quote
-  note left
-    quote is defined with reportbody =
-    Hash(<CC_NAME, CC_VER, Org_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, TLCC_MR, CH_id>)
+group enclave attestation
+  note over E_Manager #FFaa00
+    As the EPID attestation requires the SPID, we let the E_Manager provide it.
+    In a concrete implementation, the E_Manager could explicitly send the SPID to ECC_Enclave.
+    Also, the SPID could be made an optional argument within an Aux_Data field.
+    This would allow for the generalization of the attestation, to enable DCAP.
   end note
+  Enclave1-> E_Manager ++: getSPID
+  return SPID
 
-  Enclave1 -> Enclave1 : seal state
-  note left
-    sealed state includes ...
-    CC_NAME, CC_VER,
-    Org_1, SIG_Org_1,
-    ECC_S_PK, ECC_S_SK,
-    ECC_E_PK, ECC_E_SK,
-    TLCC_MR, CH_id
+  Enclave1 -> Enclave1 : Attestation <- attest(SPID, Attested_Data)
+  note left #FFaa00
+    quote is defined with reportbody = Hash(<Attested_Data>)
   end note
 
-  return ECC_S_PK, ECC_E_PK, sealed state, quote
-
-  return ECC_S_PK, ECC_E_PK, sealed state, quote, SIG_Org_1
+  note over Enclave1 #FFaa00
+    Ideally the enclave itself is able to retrieve the verifiable evidence
+    (i.e., the verification report, using EPID) and return it.
+    This additionally requires some ocalls and helper code in untrusted environment.
+    For convenience, the enclave now returns the attestation, and the evidence
+    (i.e., the verification report) is retrieved from untrusted environment.
+  end note
+  Enclave1 -> IAS ++: Attestation
+  return Evidence <- Verification_Report
+end group
 
-  |||
+  group Persistent Credentials (to be implemented)
+    Enclave1 -> Enclave1 : Sealed_Credentials <- seal(CC_Parameters, Private_Keys)
+  end group
 
-  Manager1 -> storage1 ++ : store sealed state
-  note right
-    sealed state is stored under a
-    unique identifier consisting of
-    CC_NAME, CC_VER, CC_ID, ECC_S_PK, ECC_E_PK
-  end note
-  return
+  Enclave1 -> Enclave1 : Credentials <- <Verb="CREATE", Attested_Data, Evidence>
```

Comment:
i didn't get an error in running puml but it showed `<Verb="CREATE", Attested_Data, Evidence>` verbatim and i couldn't make any sense of the Verb="CREATE" and thought that might be some puml-ism which went wrong.  I see now that you do have a verb attribute in `Credential` but i have zero intuition what the semantics is and why we have that.  If we really need that, can we give it a more semantically-meaningful name and given that it seems to be an enum, also define it as an enumeration?

Reply:
it's a constant string - I have added a comment

Reply:
would be more useful to add the comment in the components file? Also why call it `verb`?  That doesn't provide any intuition from what you added as comment.  And from that comment i still do understand why we need it in the first place? What does "reuse a protobuf mean"?

Reply:
> would be more useful to add the comment in the components file? 

if somebody is unhappy looking at `verb` it can look it up and will find the comment.

> Also why call it verb?

Because it's a "verb", and a convention we have successfully used in PDO.

> That doesn't provide any intuition from what you added as comment.

I have explained in the comment what and why.

> And from that comment i still do understand why we need it in the first place?

(we discussed this several times, but here we go again)
The expectation is that an admin will request re-attestations in the future. These will not go through the `create` but rather through another api. Yet, the attestation block will be the same (or similar). The verb used in the re-attestation will be different, so we can distinguish the two cases.
 
> What does "reuse a protobuf mean"?

It means that the protobuf message will be generated "also" in other parts of the code. So, we will not change the protobuf, but rather simply the verb.

Reply:
> Because it's a "verb", and a convention we have successfully used in PDO.

well, you could replace `Credentials' by `Noun` and it would also be correct but not very intuitive.   The fact that it is used in PDO doesn't help anybody who is look at FPC to understand it.

What about just calling it version (with the first version having to be the one created when the seal-key is created)

> (we discussed this several times, but here we go again)
The expectation is that an admin will request re-attestations in the future. These will not go through the create but rather through another api. Yet, the attestation block will be the same (or similar). The verb used in the re-attestation will be different, so we can distinguish the two cases.

Then why not just write exactly this in the comment?  I agree that we have to distinguish the initial version from later ones but lets describe it with a more meaningful name and let's indicate the issue explicitly in the comment?  I knew the issue (i think i even brought it up in the first place) but the fact that i couldn't connect the dots makes it kind of obvious that the name is not intuitive?

Reply:
af26f472796ebeaf9248fc028485010daae83a7e
yes, you brought it up in the first place